core/src/ast/expr.rs

```diff
@@ -94,6 +94,18 @@ impl ToSqlUnquoted for Expr {
     }
 }
 
+impl<'a> TryInto<&'a str> for &'a Expr {
+    type Error = ();
+
+    fn try_into(self) -> Result<&'a str, Self::Error> {
+        match self {
+            Expr::Identifier(s) => Ok(s),
+            Expr::CompoundIdentifier { ident, .. } => Ok(ident),
+            _ => Err(()),
+        }
+    }
+}
+
```

Comment:
this one looks that primary_key planner is currently new to use this.
but could we have better idea not to add this kind of TryInto conversion trick?
it might better not to have this.

Reply:
Refactored code to remove try_into all together in a clean manner.